src/utils/SortingNetworkGenerator.java

```diff
@@ -1,136 +1,273 @@
+                        for (double pos : group.values()) {
+                            if (pos > w) {
+                                w = pos;
+                            }
+                        }
+                        w += xScale;
+                        group.clear();
+                        break;
+                    }
+                }
+                double cx = w;
+                for (Entry<Comparator, Double> entry : group.entrySet()) {
+                    Comparator other = entry.getKey();
+                    double otherPos = entry.getValue();
+                    if (otherPos >= cx && c.overlaps(other)) {
+                        cx = otherPos + xScale / 3.0;
+                    }
+                }
+                group.put(c, cx);
+
+                if ((++progress & 1023) == 0) {
+                    monitor.setProgress(progress);
+                    if (monitor.isCanceled()) return true;
+                }
+            }
+            group.clear();
+            monitor.setNote("Writing SVG");
+            out.write("<svg width='" + (w + xScale) + "' height='" + h + "' xmlns='http://www.w3.org/2000/svg'>");
+            w = xScale;
+
+            writer = new WriterBuilderProxy(out);
         }
-        SortingNetworkFetcher fetcher = new SortingNetworkFetcher();
-        try {
-            ProcessBuilder builder = new ProcessBuilder(pythonCommand, "-c",
-                new BufferedReader(new InputStreamReader(fetcher.getStream())).lines().collect(Collectors.joining("\n")),
-                "--svg", path);
-            builder.redirectOutput(Redirect.INHERIT);
-            Process p = builder.start();
-            BufferedWriter w = new BufferedWriter(new OutputStreamWriter(p.getOutputStream()));
-            w.write(result);
-            w.close();
-            if (p.waitFor() != 0) {
-                BufferedReader r = new BufferedReader(new InputStreamReader(p.getErrorStream()));
-                JErrorPane.invokeCustomErrorMessage(r.lines().collect(Collectors.joining("\n")));
-                return false;
+
+        for (Comparator c : comparators) {
+            for (Comparator other : group.keySet()) {
+                if (c.hasSameInput(other)) {
+                    for (double pos : group.values()) {
+                        if (pos > w) {
+                            w = pos;
+                        }
+                    }
+                    w += xScale;
+                    group.clear();
+                    break;
+                }
+            }
+
+            double cx = w;
+            for (Entry<Comparator, Double> entry : group.entrySet()) {
+                Comparator other = entry.getKey();
+                double otherPos = entry.getValue();
+                if (otherPos >= cx && c.overlaps(other)) {
+                    cx = otherPos + xScale / 3.0;
+                }
+            }
+
+            int y0 = (c.i1 + 1) * yScale;
+            int y1 = y0;
```

Comment:
isn't `y1` supposed to be in terms of `c.i2`?

Reply:
Yes, it is. That was causing a bug. Fixed in https://github.com/Gaming32/ArrayV-v4.0/pull/74/commits/ac6ffaa1d04e85543d2a6ed51ad52a81dba56c9e.